Doc/reference/expressions.rst

```diff
@@ -1221,9 +1225,11 @@ operators and one for additive operators:
 
 The ``*`` (multiplication) operator yields the product of its arguments.  The
 arguments must either both be numbers, or one argument must be an integer and
-the other must be a sequence. In the former case, the numbers are converted to a
-common type and then multiplied together.  In the latter case, sequence
-repetition is performed; a negative repetition factor yields an empty sequence.
+the other must be a sequence, or at least one of them must be a custom object
+implementing the respective :meth:`__mul__` or :meth:`__rmul__` method. In the
+first case, the numbers are converted to a common type and then multiplied
+together.  In the second case, sequence repetition is performed; a negative
+repetition factor yields an empty sequence.
```

Comment:
I see what you're trying to say here but this became less readable now. Former/latter is preferred over first/second. More importantly, when you now mention custom objects with `__mul__` or `__rmul__`, the sentence after seems like it's talking about *those methods* and not the either-or.

Instead of this, I would do what you do in other places: add a single-sentence paragraph saying that the operation can be customized.